SADEAFxSMU/sadeaf-app
# Client Registration (Create Profile) Form

# What changes are made in this PR?
Implements the Create Profile / New User Registration feature described in #93 
After successfully registering and verifying email via cognito, the user should visit /registration to sign up as one of the roles (client, volunteer or service_requestor) and fill up their profile info.

Resolves partially #93

![image](https://user-images.githubusercontent.com/31790206/97305382-fab18b80-1897-11eb-8c24-4b5223a07bc1.png)

# Caveats
- Volunteer and Service Requestor forms need to be done aft this, but this is the baseline.

# How has this been tested?
- Log in as sadeaf-user, you should be redirected to /registration (since role is 'pending' at this stage)
- Click the 'Client' button to register as a client
- The form is shown. Fill it up and submit. You should be redirected to /pending.
- Open incognito, login as admin and approve the new user (who's registering as a client).
- The client has been onboarded successfully

# Checklist
- [x] New and existing tests pass successfully
